### PR TITLE
Update crictl to v1.12.0

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Update crictl to v1.12.0: https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.12.0.

cc @kubernetes/sig-node-pr-reviews @kubernetes/sig-cluster-lifecycle-pr-reviews
@kubernetes/sig-gcp-pr-reviews

/assign @Random-Liu @tpepper

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Update crictl to v1.12.0
```

### CODE REVIEW

**Comment by tpepper** [comment]
Test failing because of https://storage.googleapis.com/kubernetes-release/crictl/crictl-v1.12.0-linux-amd64 missing.

**Comment by tpepper** [comment]
@kubernetes/sig-node-bugs is this a release blocker?

**Comment by feiskyer** [comment]
@Random-Liu Could you help to update the file?

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -28,8 +28,8 @@ DEFAULT_CNI_VERSION="v0.6.0"
 DEFAULT_CNI_SHA1="d595d3ded6499a64e8dac02466e2f5f2ce257c9f"
 DEFAULT_NPD_VERSION="v0.5.0"
 DEFAULT_NPD_SHA1="650ecfb2ae495175ee43706d0bd862a1ea7f1395"
-DEFAULT_CRICTL_VERSION="v1.11.1"
-DEFAULT_CRICTL_SHA1="527fca5a0ecef6a8e6433e2af9cf83f63aff5694"
+DEFAULT_CRICTL_VERSION="v1.12.0"
+DEFAULT_CRICTL_SHA1="6c8540bff581970344d170ad2e785c5dafee5e1d"
```
Comment (Random-Liu) [inline cluster/gce/gci/configure.sh]: `82ef8b44849f9da0589c87e9865d4716573eec7f`

**Comment by tpepper** [comment]
At its current rate this is unlikely to make 1.12.0.  Without a bit of heroics it's going to be 1.12.1.  Specifically:

First, still looking for approvers:
* build/OWNERS (eg: @ixdy)
* cluster/gce/gci/OWNERS (eg: @dchen1107 @filbranden @yguo0905 )

Second, while the post-thaw merging through tide is progressing it still probably has a day to so this change wouldn't merge until late today or early tomorrow which is our release day (after delaying it).  Given this was found late, it wouldn't have merged for 1.12.0 on the original planned release day.  That means it's not in the master branch to be a cherry pick candidate for release-1.12 branch.

I see three options here:
* Delay the release again until this is merged and cherry picked.  I'm not inclined to do this.
* It's unfortunate, but since it would've already missed 1.12.0 in the original schedule, if we don't merge this, we fall back to marking it as a Known Issue in the 1.12.0 release note in https://github.com/kubernetes/kubernetes/issues/68840.  This is the most realistic option in my opinion.
* If it is viewed as a release blocker, if it is merge approved here and just in the backlog for master, we could consider doing a cherry pick of the commit into release-1.12 branch but from feiskyer/kubernetes instead of from master.  If folks want to consider this route, it should happen ASAP like immediately, so we still get a day's worth of CI on release-1.12 branch.

Pending clear direction to the 1.12 release team from the stakeholders here that we must do otherwise and folks clearly declaring one of the options above as the chosen path, my assumption moving forward is that this will be 1.12.1.

**Comment by dims** [comment]
We need to make changes https://github.com/kubernetes/release/search?q=cri-tools&unscoped_q=cri-tools too!

**Comment by ixdy** [comment]
it looks like https://github.com/kubernetes/release/pull/641 is the corresponding change in k/release, though it needs version-gating.

**Comment by tpepper** [comment]
@feiskyer if you want this in the Known Issues for 1.12.0, please put it in #https://github.com/kubernetes/kubernetes/issues/68840

**Comment by feiskyer** [comment]
ping @dchen1107 @yguo0905 for gci approval

**Pull request merged**